### ISSUE TITLE
Variable insight for std::vector variables

### ISSUE THREAD

**Issue by RobertoRoos** [Marus/cortex-debug#734]
I cannot see the content of `std::vector<T>` type variables. Is that a limitation by Cortex Debug or by VS Code itself?

For me it would be very useful to see the values inside vectors.

Interestingly I see that `std::array<T, N>` variables can be viewed fully.

Screenshot of my debugger:
![image](https://user-images.githubusercontent.com/12717020/187860128-f3c59be8-6f65-4fb4-ba16-5aa8b2515af5.png)
`fifo_data` is of type `std::array<uint8_t, 263>` and `impedances_mag` is `std::vector<float>`.

**Comment by haneefdm** [comment]
I don't know. Could you do me a favor and use cppdbg and write a small program for a native desktop and see if they can do that. cppdbg ms Micosts c++ debugger in cpptools

https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools

If they can do it, we can see how they did and try to something similar. We just give them a bunch of data to VSCode and they interpret/render it.

**Comment by haneefdm** [comment]
I hate to ask for this, but could you do me a favor and look at this and vote for the issue on the MS Website if you agree.

https://github.com/Marus/cortex-debug/issues/729#issuecomment-1233293715

**Comment by RobertoRoos** [comment]
Good point. This is the debugger result of `std::vector` on VS code running in Windows using GDB:

![image](https://user-images.githubusercontent.com/12717020/188381732-41baea3e-c973-4094-b12f-e77705932ca9.png)

So this is not much better.

I can properly see the vector when I build with VS 2022 and use the VS debugger (debug type "Windows" in VS Code, instead of "cppdbg")

And I voted for the issue you linked to.

**Comment by haneefdm** [comment]
No, you can't compare VS 2022 to VSCode. VS is proprietary and does not have to follow any rules and VSCode (us and cppdbg) have to follow a spec that serves as a contract between a debug-adapter like ours (and cppdbg) and VSCode core.

**Comment by haneefdm** [comment]
I also cannot observe that VS does because it is all close sourced and I have zero access to anything they do.

**Comment by ehntoo** [comment]
I believe the display of standard library types is handled by GDB. If your GDB supports python, you can add GDB's STL pretty-printers with `register_libstdcxx_printers`: https://sourceware.org/gdb/wiki/STLSupport https://gist.github.com/daverigby/99dae6997bced43a8fcdbfcb15e36116. The pretty-printers should already be included with most distributions of GDB.

**Comment by haneefdm** [comment]
I don't see a good solution for this and it is not on our roadmap.

Closing this for now.
